dotnet/samples/Concepts/Agents/ChatCompletion_Rag.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.Connectors.InMemory;
+using Microsoft.SemanticKernel.Data;
+
+namespace Agents;
+
+#pragma warning disable SKEXP0130 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+/// <summary>
+/// Demonstrate creation of <see cref="ChatCompletionAgent"/> and
+/// adding simple retrieval augmented generation (RAG) capabilities to it.
+/// </summary>
+/// <remarks>
+/// This example shows how to use the <see cref="TextSearchStore{TKey}"/> class which is designed
+/// to simplify the process of storing and searching text documents by having a built in schema.
+/// If you want to control the schema yourself, you can use an implementation of <see cref="IVectorStoreRecordCollection{TKey, TRecord}"/>
+/// with the <see cref="VectorStoreTextSearch{TRecord}"/> class instead.
+/// </remarks>
+public class ChatCompletion_Rag(ITestOutputHelper output) : BaseTest(output)
+{
+    private const string AgentName = "FriendlyAssistant";
+    private const string AgentInstructions = "You are a friendly assistant";
+
+    /// <summary>
+    /// Shows how to do Retrieval Augmented Generation (RAG) with some basic text strings.
+    /// </summary>
+    [Fact]
+    private async Task BasicRagAsync()
```

Comment:
Something more descriptive without `Async` suffix.

```suggestion
    private async Task UseChatAgentWith...()
```

Reply:
The agent is specifically called ChatCompletionAgent, not ChatAgent, so I'll update to UseChatCompletionAgentWithBasicRag.